airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/state_converters/datetime_stream_state_converter.py

```diff
@@ -17,17 +17,13 @@
 
 class DateTimeStreamStateConverter(AbstractStreamStateConverter):
 
-    def __init__(self, start: Optional[Any], cursor_start: Optional[Any]):
-        start = self.parse_timestamp(start) or self.zero_value
-        self.prev_sync_low_water_mark = self.parse_timestamp(cursor_start) if cursor_start else None
-        if self.prev_sync_low_water_mark and self.prev_sync_low_water_mark >= start:
-            self._actual_sync_start = self.prev_sync_low_water_mark
+    def get_sync_start(self, cursor_field: CursorField, stream_state: MutableMapping[str, Any], start: Optional[Any]) -> datetime:
+        sync_start = self.parse_timestamp(start) or self.zero_value
+        prev_sync_low_water_mark = self.parse_timestamp(stream_state[cursor_field.cursor_field_key]) if cursor_field.cursor_field_key in stream_state else None
+        if prev_sync_low_water_mark and prev_sync_low_water_mark >= sync_start:
+            return prev_sync_low_water_mark
         else:
-            self._actual_sync_start = start
```

Comment:
Should we log `if prev_sync_low_water_mark and  prev_sync_low_water_mark < sync_start`?